src/Components/Authorization/Mobile/RegisterForm.tsx

```diff
@@ -12,25 +13,53 @@ import Text from "../../Text"
 import TextLink from "../../TextLink"
 import Colors from "Assets/Colors"
 import { Validators } from "../Validators"
+import { metaphysics } from "../../../Utils/metaphysics"
+import * as sharify from "sharify"
+
+const checkEmail = (values, actions) => {
+  const query = `
+    query {
+      user(email: "${values.email}") {
+        userAlreadyExists
+      }
+    }
+  `
+
+  return metaphysics(
+    { query },
+    {
+      appToken: sharify.data.XAPP_TOKEN,
+    }
+  ).then(({ data }: any) => {
+    if (data.user.userAlreadyExists) {
+      actions.setFieldError("email", "Email already exists.")
+      actions.setSubmitting(false)
+      return false
+    }
+    return true
+  })
+}
 
 export const MobileRegisterForm: FormComponentType = props => {
   const steps = [
-    <Step>
+    <Step validationSchema={Validators.email} onSubmit={checkEmail}>
       {({
         wizard,
-        form: { errors, touched, values, handleChange, handleBlur },
+        form: { errors, touched, values, handleChange, handleBlur, setTouched },
       }) => (
         <div>
           <Input
             block
-            error={touched.email && errors.email}
+            error={errors.email}
```

Comment:
Do we want the error to show immediately rather than after focus is lost? I noticed in the screen cap that as soon as you begin typing you get validation errors. You might also be able to use the new Artsy Input wrapped in a Formik Field at https://github.com/artsy/reaction/blob/master/src/Components/Forms/support/Field.tsx